contracts/MetaGuard.sol

```diff
@@ -0,0 +1,187 @@
+    function checkTransaction(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver,
+        bytes memory signatures,
+        address msgSender
+    ) external override {
+        address currentGuard = guards[SENTINEL_GUARDS];
+        while (currentGuard != SENTINEL_GUARDS) {
+            IGuard(currentGuard).checkTransaction(
+                to,
+                value,
+                data,
+                operation,
+                safeTxGas,
+                baseGas,
+                gasPrice,
+                gasToken,
+                refundReceiver,
+                signatures,
+                msgSender
+            );
+            currentGuard = guards[currentGuard];
+        }
+    }
+
+    function checkAfterExecution(bytes32 txHash, bool success)
+        external
+        override
+    {
+        address currentGuard = guards[SENTINEL_GUARDS];
+        while (currentGuard != SENTINEL_GUARDS) {
+            IGuard(currentGuard).checkAfterExecution(txHash, success);
+            currentGuard = guards[currentGuard];
+        }
+    }
+
+    /// @dev Removes a guard on meta guard.
+    /// @notice This can only be called by the owner.
+    /// @param prevGuard Guard that pointed to the guard to be removed in the linked list.
+    /// @param guard Guard to be removed.
+    function removeGuard(address prevGuard, address guard) public onlyOwner {
+        if (guard == address(0) || guard == SENTINEL_GUARDS)
+            revert InvalidGuard(guard);
+        if (guards[prevGuard] != guard) revert AlreadyRemovedGuard(guard);
+        guards[prevGuard] = guards[guard];
+        guards[guard] = address(0);
+        emit RemovedGuard(guard);
+    }
+
+    /// @dev Adds a guard on meta guard.
+    /// @param guard Address of the guard to be addd.
+    /// @notice This can only be called by the owner.
+    function addGuard(address guard) public onlyOwner {
+        if (guard == address(0) || guard == SENTINEL_GUARDS)
+            revert InvalidGuard(guard);
+        if (guards[guard] != address(0)) revert AlreadyAddedGuard(guard);
+        guards[guard] = guards[SENTINEL_GUARDS];
+        guards[SENTINEL_GUARDS] = guard;
+        emit AddedGuard(guard);
+    }
+
+    /// @dev Returns if an guard is added
+    /// @return True if the guard is added
+    function isGuardAdded(address _guard) public view returns (bool) {
+        return SENTINEL_GUARDS != _guard && guards[_guard] != address(0);
+    }
+
+    /// @dev Returns array of guards.
+    /// @param start Start of the page.
+    /// @param pageSize Maximum number of guards that should be returned.
+    /// @return array Array of guards.
+    /// @return next Start of the next page.
+    function getGuardsPaginated(address start, uint256 pageSize)
```

Comment:
I'm curious why this is paginated--is there a reason we don't just return all of them?

Reply:
There is no particular reason actually just following how the zodiac returns the module, I will add the get all function as well I think

Reply:
Personally I’d just suggest a get all and not even bother with a paginated request. Pagination seems like a client concern.

Reply:
get all function added